test/library/idpdeployment.go

```diff
@@ -71,19 +81,75 @@ func deployPod(
 		metav1.CreateOptions{},
 	)
 
+	saName := name
```

Comment:
**nit:** if you want to create a copy of the name for the deployment and use it deliberately as the name of the ServiceAccount, I would do it above the creation of the ServiceAccount and use it in there:

```go
	saName := name
	_, err = clients.CoreV1().ServiceAccounts(namespace).Create(
		testContext,
		&corev1.ServiceAccount{
			ObjectMeta: metav1.ObjectMeta{
				Name: saName,
			},
		},
		metav1.CreateOptions{},
	)
```

Reply:
Good point, we missed this in the original PR -- but for the sake of consistency with the original (#679) let's not change it in the backport, WDYT?